src/ScheduledTasks/Entity/Schedule.cs

```diff
@@ -0,0 +1,301 @@
+        }
+        catch (Exception ex)
+        {
+            this.logger.ScheduleOperationError(scheduleCreationOptions.ScheduleId, nameof(this.CreateSchedule), "Failed to create schedule", ex);
+            throw;
+        }
+    }
+
+    /// <summary>
+    /// Updates an existing schedule.
+    /// </summary>
+    /// <param name="context">The task entity context.</param>
+    /// <param name="scheduleUpdateOptions">The options for updating the schedule configuration.</param>
+    /// <exception cref="ArgumentNullException">Thrown when scheduleConfigUpdateOptions is null.</exception>
+    /// <exception cref="InvalidOperationException">Thrown when the schedule is not created.</exception>
+    public void UpdateSchedule(TaskEntityContext context, ScheduleUpdateOptions scheduleUpdateOptions)
+    {
+        try
+        {
+            if (!this.CanTransitionTo(nameof(this.UpdateSchedule), this.State.Status))
+            {
+                throw new ScheduleInvalidTransitionException(this.State.ScheduleConfiguration?.ScheduleId ?? string.Empty, this.State.Status, this.State.Status, nameof(this.UpdateSchedule));
+            }
+
+            if (scheduleUpdateOptions == null)
+            {
+                throw new ScheduleClientValidationException(this.State.ScheduleConfiguration?.ScheduleId ?? string.Empty, "Schedule update options cannot be null");
+            }
+
+            Verify.NotNull(this.State.ScheduleConfiguration, nameof(this.State.ScheduleConfiguration));
+
+            HashSet<string> updatedScheduleConfigFields = this.State.ScheduleConfiguration.Update(scheduleUpdateOptions);
+            if (updatedScheduleConfigFields.Count == 0)
+            {
+                // no need to interrupt and update current schedule run as there is no change in the schedule config
+                this.logger.ScheduleOperationDebug(this.State.ScheduleConfiguration.ScheduleId, nameof(this.UpdateSchedule), "Schedule configuration is up to date.");
+                return;
+            }
+
+            this.State.ScheduleLastModifiedAt = DateTimeOffset.UtcNow;
+
+            // after schedule config is updated, perform post-config-update logic separately
+            foreach (string updatedScheduleConfigField in updatedScheduleConfigFields)
+            {
+                switch (updatedScheduleConfigField)
+                {
+                    case nameof(this.State.ScheduleConfiguration.StartAt):
+                    case nameof(this.State.ScheduleConfiguration.Interval):
+                        this.State.NextRunAt = null;
+                        break;
+
+                    // TODO: add other fields's callback logic after config update if any
+                    default:
+                        break;
+                }
+            }
+
+            this.State.RefreshScheduleRunExecutionToken();
+
```

Comment:
Don't we need a `context.SignalEntity(...` here also? 

The previous signal is no longer valid after we change the execution token.

Reply:
should signal if status is active